### ISSUE TITLE
Multiple folders reported as flat structure in HTML report and causing failure with --html-details

### ISSUE THREAD

**Issue by mambu** [gcovr/gcovr#126]
I am using gcovr to produce coverage reports for a project that is structured in this way:

Source code
- $ROOT/src/lib1/
- $ROOT/src/lib2/

Build files (obj files and gcov files):
- $ROOT/debug64/obj/lib1.a/
- $ROOT/debug64/obj/lib2.a/

Running `gcovr -v -k -s -r . --html --html-details -o fhcore.html` I in $ROOT get the following error:

```
Running gcov: 'gcov /home/mambu/temp/gcov-fhcore/debug64/obj/libFhUDP.a/FhUDPOrderedMsgCacher.gcno --branch-counts --branch-probabilities --preserve-paths --object-directory /home/mambu/temp/gcov-fhcore/debug64/obj/libFhUDP.a' in '/home/mambu/temp/gcov-fhcore/debug64/obj/libFhUDP.a'
Gathered coveraged data for 125 files
Traceback (most recent call last):
  File "/usr/bin/gcovr", line 1970, in <module>
    print_html_report(covdata, options.html_details)
  File "/usr/bin/gcovr", line 1473, in print_html_report
    INPUT = open(data['FILENAME'], 'r')
IOError: [Errno 2] No such file or directory: 'FhDsDepthEventPublisher.h'
```

Running the same command line in $ROOT without parameter `html-details` works OK:
`gcovr -v -k -s -r . --html -o fhcore.html`

```
Gathered coveraged data for 125 files
lines: 30.6% (2503 out of 8168)
branches: 16.0% (1358 out of 8484)
```

Basically the issue seems to be that the file `src/lib1/FhDsDepthEventPublisher.h` is looked for in $ROOT (.) folder, rather than in `$ROOT/src/lib1/`.

Also HTML file generated without the details parameter contains a flat structure. That is I see the coverage for `FhDsDepthEventPublisher.h`, not for `src/lib1/FhDsDepthEventPublisher.h`. All the source files are listed in without their folder.

Am I doing something wrong? Is this scenario supported by gcovr 3.2?

Thanks

**Comment by Russtopia** [comment]
I have the same issue. Running gcovr at the top of a hierarchy of dirs with source files in each with -html-details results in only one .html file at the top, with a flat list of all source files in the deep tree; and the hyperlinks for source files residing below the html file's dir are wrong (no sub-dir path leading to the actual .html file, which isn't generated anyways).

I also see the [Errno 2] No such file or directory: errors during such a top-level analysis.

For expected results, compare to the behaviour and output of the lcov suite:

lcov -c -d . -o lcov.dat
genhtml lcov.dat

This properly generates a top-level and all subdir html reports with links allowing full navigation to subdirs and source file coverage detail HTMl pages.

**Comment by igagis** [comment]
Steps to reproduce:
1. `git clone --recurse-submodules https://github.com/cppfw/utki`
1. `cd utki`
1. `mkdir coverage`
1. `make config=gcov test`
1. `gcovr -r src --json > src/cov.json`
1. `gcovr -r tests/unit > tests/unit/cov.json`
1. `gcovr -a src/cov.json -a tests/unit/cov.json --html --html-details coverage/report.html`

Result at the last step:
```console
ivan@igagisw:~/prj/utki$ gcovr -a src/cov.json -a tests/unit/cov.json --html --html-details coverage/report.html
Traceback (most recent call last):
  File "/usr/bin/gcovr", line 11, in <module>
    load_entry_point('gcovr==4.2', 'console_scripts', 'gcovr')()
  File "/usr/lib/python3/dist-packages/gcovr/__main__.py", line 253, in main
    print_reports(covdata, options, logger)
  File "/usr/lib/python3/dist-packages/gcovr/__main__.py", line 368, in print_reports
    generator(covdata, output.value, options)
  File "/usr/lib/python3/dist-packages/gcovr/html_generator.py", line 248, in print_html_report
    with io.open(data['FILENAME'], 'r', encoding=options.source_encoding,
FileNotFoundError: [Errno 2] No such file or directory: 'src/debug.cpp'
```

**Comment by igagis** [comment]
Steps to reproduce in Debian/Ubuntu:
1. Setup [prorab-extra](https://github.com/cppfw/prorab-extra) (a build system used by the sample project)
1. `git clone --recurse-submodules https://github.com/cppfw/utki`
1. `cd utki`
1. `git checkout 462be08cb47be77c2af2b2669a9a0367a959dcc5`
1. `mkdir coverage`
1. `make config=gcov test`
1. `gcovr -r src --json > src/cov.json`
1. `gcovr -r tests/unit --json > tests/unit/cov.json`
1. `gcovr -a src/cov.json -a tests/unit/cov.json --html --html-details coverage/report.html`

Result at the last step:
```console
ivan@igagisw:~/prj/utki$ gcovr -a src/cov.json -a tests/unit/cov.json --html --html-details coverage/report.html
Traceback (most recent call last):
  File "/usr/bin/gcovr", line 11, in <module>
    load_entry_point('gcovr==4.2', 'console_scripts', 'gcovr')()
  File "/usr/lib/python3/dist-packages/gcovr/__main__.py", line 253, in main
    print_reports(covdata, options, logger)
  File "/usr/lib/python3/dist-packages/gcovr/__main__.py", line 368, in print_reports
    generator(covdata, output.value, options)
  File "/usr/lib/python3/dist-packages/gcovr/html_generator.py", line 248, in print_html_report
    with io.open(data['FILENAME'], 'r', encoding=options.source_encoding,
FileNotFoundError: [Errno 2] No such file or directory: 'src/debug.cpp'
```

**Comment by latk** [comment]
@igagis Your issue is related to the merging of JSON coverage data, not to the HTML writer. As documented for the JSON output format, the source file paths in the JSON file are relative to the root directory. Your example features JSON files with different root directories, so they cannot be merged. See also #397.

**Comment by Spacetown** [comment]
Since there is no progress for a while IÄ'm closing this issue. Please reopen if it still persists.
